app/questionnaire/questionnaire_store_updater.py

```diff
@@ -13,15 +14,26 @@ def __init__(self, current_location, schema, questionnaire_store, current_questi
         self._schema = schema
         self._questionnaire_store = questionnaire_store
         self._answer_store = self._questionnaire_store.answer_store
+        self._list_store = self._questionnaire_store.list_store
 
-    def save_answers(self, form):
+    def save_answers(self, form, save_completed_blocks=True):
         self._update_questionnaire_store_with_form_data(form.data)
 
-        if self._current_location not in self._questionnaire_store.completed_blocks:
-            self._questionnaire_store.completed_blocks.append(self._current_location)
+        if save_completed_blocks:
```

Comment:
I don't get why?

Reply:
We don't want to add blocks which are not on the routing path to completed blocks.